src/get-version.ts

```diff
@@ -1,12 +1,18 @@
 import * as fs from 'fs-extra'
 import HTTP from 'http-call'
 
-async function run(name: string, file: string, version: string) {
-  await fs.outputJSON(file, {current: version}) // touch file with current version to prevent multiple updates
-  const {body} = await HTTP.get<any>(`https://registry.npmjs.org/${name.replace('/', '%2f')}`, {timeout: 5000})
-  await fs.outputJSON(file, {...body['dist-tags'], current: version})
+async function run(name: string, file: string, version: string, registry: string, authorization: string) {
+  const url = [
+    registry.replace(/\/+$/, ''), // remove trailing slash
+    name.replace('/', '%2f')      // scoped packages need escaped separator
+  ].join('/')
+  const headers = authorization ? {authorization} : {}
+
+  await fs.outputJSON(file, {current: version, headers}) // touch file with current version to prevent multiple updates
+  const {body} = await HTTP.get<any>(url, {headers, timeout: 5000})
+  await fs.outputJSON(file, {...body['dist-tags'], current: version, authorization})
   process.exit(0)
 }
 
-run(process.argv[2], process.argv[3], process.argv[4])
+run(process.argv[2], process.argv[3], process.argv[4], process.argv[5], process.argv[6])
```

Comment:
arguably this is hitting a point where it might be better just to fetch the config again but this is fine for now